widget/entry.go

```diff
@@ -247,6 +248,16 @@ func (e *Entry) KeyDown(key *fyne.KeyEvent) {
 		}
 		e.selectKeyDown = true
 	}
+	if key.Name == fyne.KeyEscape {
+		if cancelBtn := e.findButton(CancelButton); cancelBtn != nil {
+			cancelBtn.KeyDown(&fyne.KeyEvent{Name: fyne.KeyEnter})
```

Comment:
This looks like form code in Entry - perhaps you can find a way that better separates the concerns?

Reply:
Not sure what you mean here. Which concerns?

Reply:
Form and entry are separate. If we want to submit a form on entry press we need to figure some way for that code to exist in Form not in Entry. It’s not an easy problem to solve, but doing it this way would need form logic inside every input widget which is what I mean about not separating concerns? (https://en.m.wikipedia.org/wiki/Separation_of_concerns)